src/library/scala/MatchError.scala

```diff
@@ -19,11 +21,11 @@ package scala
  *  @version 1.1, 05/03/2004
  *  @since   2.0
  */
-final class MatchError(obj: Any) extends RuntimeException {
+final class MatchError(@transient obj: Any) extends RuntimeException {
   /** There's no reason we need to call toString eagerly,
    *  so defer it until getMessage is called.
    */
-  private lazy val objString = {
+  private val objString = {
```

Comment:
@hrhino ... you're absolutely right. I just started implementing it one way and then didn't finish thinking through how the other changes could make it easier.

There's no need to reimplement lazy vals :)